src/python.cpp

```diff
@@ -612,9 +650,18 @@ SEXP py_to_r(PyObject* x, bool convert) {
       return LogicalVector::create(x == Py_True);
 
     // integer
-    else if (scalarType == INTSXP)
-      return IntegerVector::create(PyInt_AsLong(x));
-
+    else if (scalarType == INTSXP) {
+      long val = PyLong_AsLong(x);
+      if((val > std::numeric_limits<int>::max() || val < std::numeric_limits<int>::min()) && convertLongToBit64()) {
```

Comment:
I worry a bit that this could be confusing if values were converted to `integer64` only "sometimes". That is, I'm not sure if the runtime type should depend on the runtime value. Would it make sense to _always_ make this conversion if the user has opted in?